### PR TITLE
Rate limit transaction counters

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Metric counters are getting dropped during tps benchmarks

#### Summary of Changes
* allow setting a high metrics rate via env var
* default high metrics rate to 1000
* introduce new macros for high rate counters

Fixes https://github.com/solana-labs/solana/issues/5422

### CODE REVIEW

**Comment by jstarry** [comment]
@sakridge I'm not very familiar to how benches are run but I want to set a high tx metrics rate value for when you're doing bench testing. Where is a good place to set an env var for that?

**Comment by sakridge** [comment]
@jstarry I don't think the env variable is a good approach. It should be able to handle any transaction rate.

**Comment by jstarry** [comment]
fixing tests...

File: multinode-demo/common.sh [hunk 1]
```diff
@@ -59,6 +59,7 @@ solana_wallet=$(solana_program wallet)
 solana_replicator=$(solana_program replicator)
 
 export RUST_BACKTRACE=1
+export SOLANA_METRICS_HIGH_RATE=10
```
Comment (sakridge) [inline multinode-demo/common.sh]: Don't you want a default of 10 and then set to 1000 here?

Comment (jstarry) [reply multinode-demo/common.sh]: Hmm I guess I'm not sure 🤔Can you help me understand the different ways a cluster is run? Do you use `multinode-demo` to start your benchmark tests?

Comment (jstarry) [reply multinode-demo/common.sh]: I just swapped them, but still want to understand to make sure all the scenarios are covered. I'm worried that having a default of `1000` in `multinode-demo` will be surprising to other engineers

Comment (sakridge) [reply multinode-demo/common.sh]: @jstarry I use both local testing with:
```
./multinode-demo/setup.sh
./multinode-demo/bootstrap-leader.sh
./multinode-demo/client.sh
```

and with a remote cluster with:

`./net/net.sh restart -f cuda`

Comment (sakridge) [reply multinode-demo/common.sh]: Also like this: `cd bench-tps; cargo test --release bench_tps_local_cluster_solana`

But rarely would i have metrics enabled for that.

Comment (jstarry) [reply multinode-demo/common.sh]: @sakridge great, thanks for clarifying! Looks like `net.sh` calls into `multinode-demo` scripts so I get now why you want a high default there. I added a small change to make sure the `multinode-demo` scripts won't override `SOLANA_METRICS_HIGH_RATE` if it is already set.

**Pull request merged**